File: cases/taint/javascript/recursive-carry-negative/RecursiveCarryNegative.js
function dfb_source() { // DFB-SOURCE: recursive-carry-input
  return "tainted";
}

function dfb_sink(value) {} // DFB-SINK: recursive-carry-sink

function carry(value, depth) {
  if (depth === 0) {
    return "clean"; // DFB-WITNESS: recursive-carry-base
  }
  return carry(value, depth - 1);
}

function run() {
  dfb_sink(carry(dfb_source(), 5));
}

File: cases/taint/java/recursive-carry-negative/RecursiveCarryNegative.java
package dataflowbench.taint;

final class RecursiveCarryNegative {
    static int dfb_source() { // DFB-SOURCE: recursive-carry-input
        return 1;
    }

    static void dfb_sink(int value) { } // DFB-SINK: recursive-carry-sink

    static int carry(int value, int depth) { // DFB-WITNESS: recursive-carry-step
        if (depth == 0) {
            return 0; // DFB-KILL: recursive-carry-clean
        }
        return carry(value, depth - 1);
    }

    static void run() {
        dfb_sink(carry(dfb_source(), 5));
    }
}
